Python/getargs.c

```diff
@@ -2587,22 +2594,38 @@ _PyArg_UnpackKeywordsWithVararg(PyObject *const *args, Py_ssize_t nargs,
         else {
             current_arg = NULL;
         }
-
-        /* If an arguments is passed in as a keyword argument,
-         * it should be placed before `buf[vararg]`.
+        /*
+         * Assume that an argument at index `i` is passed in
+         * as a keyword argument. Depending on `i`, it should
+         * be placed before `buf[vararg]` or at `buf[i + 1]`.
+         *
+         * It is placed at `buf[vararg]` if `nargs < vararg`
+         * and `i < vararg`, e.g.:
+         *
+         *      def f(a, /, b, *args): pass
          *
-         * For example:
-         * def f(a, /, b, *args):
-         *     pass
-         * f(1, b=2)
+         * The `buf` array for "f(1, b=2)" is:
          *
-         * This `buf` array should be: [1, 2, NULL].
-         * In this case, nargs < vararg.
+         *      buf = [1, 2, NULL].
          *
-         * Otherwise, we leave a place at `buf[vararg]` for vararg tuple
-         * so the index is `i + 1`. */
-        if (nargs < vararg && i != vararg) {
-            buf[i] = current_arg;
+         * On the other hand, if the argument is a real
+         * keyword argument (namely `i > vararg`), then
+         * it is placed at `buf[i + 1]` since `buf[vararg]`
+         * is already taken, e.g.:
+         *
+         *      def g(a, b, *args, c, d): ...
+         *
+         * The `buf` array for "f(a=1, b=2, c=3, d=4)" is:
+         *
+         *      buf = [1, 2, NULL, 3, 4].
+         */
+        if (nargs < vararg && i < vararg) {
+            if (current_arg != NULL) {
```

Comment:
This left `buf[i]` not initialized, isn't?

Reply:
Urgh... yes. I should probably change the way it's being checked... like a boolean checking whether I replaced something before or not. But depending on whether you want to change the logic of this function, I may delay the commit.